cognite/client/_api/datapoints.py

```diff
@@ -2280,6 +2285,11 @@ def insert_dataframe(self, df: pd.DataFrame, external_id_headers: bool = True, d
                 continue
             if external_id_headers:
                 dps.append({"datapoints": datapoints, "externalId": column_id})
+            elif instance_id_headers:
+                if isinstance(column_id, NodeId):
+                    dps.append({"datapoints": datapoints, "instanceId": column_id})
+                elif isinstance(column_id, tuple):
+                    dps.append({"datapoints": datapoints, "instanceId": NodeId.load(column_id)})
```

Comment:
The `.load` method of `NodeId` can catch this already. In addition, if the user pass in `instance_id_headers` but don't have that type of headers, I think we should raise and not let it pass silently. 
```suggestion
                if isinstance(column_id, NodeId | tuple | dict):
                    dps.append({"datapoints": datapoints, "instanceId": NodeId.load(column_id)})
                else:
                    raise ValueError("Could not find instance Ids in the coulnm header. Please InstanceId are given as NodeId, tuple, or dict. Got {type(column_id)}"
```